lightning/src/chain/channelmonitor.rs

```diff
@@ -2008,33 +2008,37 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 			self.is_paying_spendable_output(&tx, height, &logger);
 		}
 
-		self.block_confirmed(height, txn_matched, watch_outputs, claimable_outpoints, broadcaster, fee_estimator, logger)
+		self.block_confirmed(height, txn_matched, watch_outputs, claimable_outpoints, &broadcaster, &fee_estimator, &logger)
 	}
 
+	/// Update state for new block(s)/transaction(s) confirmed. Note that the caller *must* update
+	/// self.best_block before calling, and set `conf_height` to the height at which any new
+	/// transaction(s)/block(s) were confirmed at, even if it is not the current best height
+	/// (assuming we are being called with new transaction(s) confirmed).
 	fn block_confirmed<B: Deref, F: Deref, L: Deref>(
 		&mut self,
-		height: u32,
+		conf_height: u32,
 		txn_matched: Vec<&Transaction>,
 		mut watch_outputs: Vec<TransactionOutputs>,
 		mut claimable_outpoints: Vec<PackageTemplate>,
-		broadcaster: B,
-		fee_estimator: F,
-		logger: L,
+		broadcaster: &B,
+		fee_estimator: &F,
+		logger: &L,
 	) -> Vec<TransactionOutputs>
 	where
 		B::Target: BroadcasterInterface,
 		F::Target: FeeEstimator,
 		L::Target: Logger,
 	{
-		let should_broadcast = self.would_broadcast_at_height(height, &logger);
+		let should_broadcast = self.would_broadcast_at_height(self.best_block.height(), logger);
```

Comment:
Hmm, suppose we could, I thought renaming height to `conf_height` was pretty clear (though I could spell it out). For `OnchainTx` I could tweak it if you think it makes more sense (given it doesn't have its own `best_block` field)?

Reply:
Seems like there are a few scenarios:
- it doesn't need to be passed at all because the method already can access it through `self`, so why bother passing it? (e.g., remove `would_broadcast_at_height` can be named simply `should_broadcast` and doesn't need to take a height, nor does `get_broadcasted_holder_claims`
- only the current height is needed, so passing `&BestBlock` makes it explicit and prevents passing the wrong height (e.g., `has_reached_confirmation_threshold`)
- both heights are needed in `OnchainTx` (maybe leave as is since there is a calculation involved at the call site)

Reply:
I believe `get_broadcasted_holder_claims` needs the confirmation height, or at least appears to given it passes it to `PackageTx::build_package` as `height_original`, though its API seems like a good opportunity for cleanup. Did the rest.

Reply:
Hmmm... I see but you're calling `get_broadcasted_holder_claims` with the best block height in `block_confirmed`, right?

Reply:
Right, in that case its being called with reference to a new commitment transaction being broadcast, not one which is confirmed (`new_holder_commitment_transaction`), so I *think* that's correct. Somewhat confusingly, its also called with a 0-height in `provide_payment_preimage` in a similar context, which is "wrong", but AFAICT harmless.

Ultimately it doesn't really matter much for this case, I think, cause its passed through as `height_original` in `PackageTemplate` which is only used to figure out when we can merge packages (which we can never do for the `HolderHTLCOutput`s that are used here), and when we can broadcast (which we just use the CLTV which is fine).

I'll add some comments and fix the 0 value version.